src/sentry/seer/anomaly_detection/utils.py

```diff
@@ -147,33 +177,47 @@ def fetch_historical_data(
         # DATASET_OPTIONS expects the name 'errors'
         dataset_label = "errors"
     elif dataset_label == "generic_metrics":
+        # XXX: this is for sessions/crash rate alerts
         dataset_label = "transactions"
+    elif dataset_label == "transactions":
+        # XXX: this is for performance alerts
+        dataset_label = "discover"
     dataset = get_dataset(dataset_label)
 
     if not project or not dataset or not alert_rule.organization:
         return None
 
+    environments = []
+    if snuba_query.environment:
+        environments = [snuba_query.environment]
+
+    snuba_params = SnubaParams(
+        organization=alert_rule.organization,
+        projects=[project],
+        start=start,
+        end=end,
+        stats_period=None,
+        environments=environments,
+    )
+    snuba_query_string = get_snuba_query_string(snuba_query)
```

Comment:
This is one of the key changes here - the front end constructs a stringified query based on `snuba_query.query` AND `snuba_query.event_types`. This adds a join to the table for things like errors count with the `is:unresolved` query, or when you're using the dropdown to select "errors", "default", or "errors OR default" event types